core/vm/jump_table.go

```diff
@@ -47,13 +47,32 @@ type operation struct {
 	// jumps indicates whether operation made a jump. This prevents the program
 	// counter from further incrementing.
 	jumps bool
+	// writes determines whether this a state modifying operation
+	writes bool
```

Comment:
Correct, because a call might not always write. This is checked [here](https://github.com/ethereum/go-ethereum/pull/14337/files#diff-8f9edb073f33b63dc5736262579687b8R97). I guess this could have been moved to the metro PR to make things more clear. 

In any case a CALL which does not transfer value (including sub calls) may be regarded as a non-writing operation.